### PR TITLE
[log] Switch out ``print()`` for ``log()``

### PR DESCRIPTION
**Pull request by lucasg**
``logging.log()`` is easier to configure and suppress than ``print()``, which is mainly done by redirecting ``stdout``.
This PR makes it easier to use ``vmlinux-to-elf`` in another python project rather than as a standalone script

### CODE REVIEW

File: vmlinux_to_elf/kallsyms_finder.py [hunk 1]
```diff
@@ -1069,15 +1069,15 @@ def print_symbols_debug(self):
             
             symbol_types.add(symbol_name[0])
         
-        print('Symbol types', '=>', sorted(symbol_types))
-        print()
+        logging.info('Symbol types => {0:s}'.format(sorted(symbol_types)))
```
Comment (marin-m) [inline vmlinux_to_elf/kallsyms_finder.py:1072]: It seems that you're passing a `list` to the `{0:s}` formatter, it's yielding an error when using the `kallsyms-finder` entry point:

```
  File "./kallsyms-finder", line 1072, in print_symbols_debug
    logging.info('Symbol types => {0:s}'.format(sorted(symbol_types)))
TypeError: unsupported format string passed to list.__format__
```

(Maybe that using `'%s' % value` for formatting is safer)

File: vmlinux_to_elf/main.py [hunk 2]
```diff
@@ -16,6 +16,8 @@
     from vmlinux_to_elf.architecture_detecter import ArchitectureGuessError
 
 if __name__ == '__main__':
+
+    logging.basicConfig(level=logging.INFO, format='%(message)s')
```
Comment (marin-m) [inline vmlinux_to_elf/main.py:20]: It seems that you forgot to add the logging level configuration to all entry points of the program (including `kallsyms_finder.py`), which causes no output to be printed when using these.

**Review by marin-m — CHANGES_REQUESTED** [review]
Hello,

Thanks for your pull request, I left a few comments.

Regards,

**Comment by lucasg** [comment]
I've pushed the fixes you proposed.

I tested with ``main.py``, ``tests.py`` and ``kallsyms-finder.py`` and it work as expected

**Comment by marin-m** [comment]
Hello,

Thanks for your changes. It seems that your script is not redirecting all the output of the tool to `stderr`, including the raw symbols constituting the primary output of the `kallsyms-finder` entry point, which is rather counterintuitive.

Maybe that keeping the existing behavior of outputting information to stdout instead of stderr would be better, in addition to preserve backwards compatibility?

Regards,

**Comment by lucasg** [comment]
Huh, I didn't know about this "feature" of ``logging``, thanks for pointing it out.

**Comment by marin-m** [comment]
Thanks, there seems to be at least one formatting issue left:

```python
--- Logging error ---
Traceback (most recent call last):
  File "/home/marin/boisse/vmlinux_to_elf/kallsyms_finder.py", line 193, in __init__
    self.find_kallsyms_token_table()
  File "/home/marin/boisse/vmlinux_to_elf/kallsyms_finder.py", line 409, in find_kallsyms_token_table
    raise KallsymsNotFoundException('%d candidates for kallsyms_token_table in kernel image' % len(candidates_offsets))
kallsyms_finder.KallsymsNotFoundException: 0 candidates for kallsyms_token_table in kernel image

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "/usr/lib/python3.6/logging/__init__.py", line 994, in emit
    msg = self.format(record)
  File "/usr/lib/python3.6/logging/__init__.py", line 840, in format
    return fmt.format(record)
  File "/usr/lib/python3.6/logging/__init__.py", line 577, in format
    record.message = record.getMessage()
  File "/usr/lib/python3.6/logging/__init__.py", line 338, in getMessage
    msg = msg % self.args
TypeError: not all arguments converted during string formatting
Call stack:
  File "./vmlinux_to_elf/tests.py", line 64, in <module>
    ElfSymbolizer(raw_data, ELF_KERNELS_OUTPUT_PATH + '/' + slugify(file_name) + '.elf')
  File "/home/marin/boisse/vmlinux_to_elf/elf_symbolizer.py", line 44, in __init__
    kallsyms_finder = KallsymsFinder(file_contents, bit_size)
  File "/home/marin/boisse/vmlinux_to_elf/kallsyms_finder.py", line 200, in __init__
    self.find_kallsyms_names_uncompressed()
  File "/home/marin/boisse/vmlinux_to_elf/kallsyms_finder.py", line 557, in find_kallsyms_names_uncompressed
    logging.info('[+] Kernel symbol names found at file offset', hex(ksymtab_match.start(0)))
Message: '[+] Kernel symbol names found at file offset'
Arguments: ('0x2b8480',)
```

**Comment by marin-m** [comment]
Merged, thanks!

**Pull request merged**